internal/provider/model_fmc_hosts.go

```diff
@@ -0,0 +1,258 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Mozilla Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://mozilla.org/MPL/2.0/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+// Section below is generated&owned by "gen/generator.go". //template:begin imports
+import (
+	"context"
+	"fmt"
+	"maps"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/tidwall/gjson"
+	"github.com/tidwall/sjson"
+)
+
+// End of section. //template:end imports
+
+// Section below is generated&owned by "gen/generator.go". //template:begin types
+
+type Hosts struct {
+	Id     types.String          `tfsdk:"id"`
+	Domain types.String          `tfsdk:"domain"`
+	Items  map[string]HostsItems `tfsdk:"items"`
+}
+
+type HostsItems struct {
+	Id          types.String `tfsdk:"id"`
+	Description types.String `tfsdk:"description"`
+	Overridable types.Bool   `tfsdk:"overridable"`
+	Ip          types.String `tfsdk:"ip"`
+	Type        types.String `tfsdk:"type"`
+}
+
+// End of section. //template:end types
+
+// Section below is generated&owned by "gen/generator.go". //template:begin getPath
+
+func (data Hosts) getPath() string {
+	return "/api/fmc_config/v1/domain/{DOMAIN_UUID}/object/hosts"
+}
+
+// End of section. //template:end getPath
+
+func (data Hosts) toBody(ctx context.Context, state Hosts) string {
+	body := "[]"
+	if len(data.Items) > 0 {
+		for key, item := range data.Items {
+			itemBody, _ := sjson.Set("{}", "name", key)
+			if !item.Id.IsNull() && !item.Id.IsUnknown() {
+				itemBody, _ = sjson.Set(itemBody, "id", item.Id.ValueString())
+			}
+			if !item.Description.IsNull() {
+				itemBody, _ = sjson.Set(itemBody, "description", item.Description.ValueString())
+			}
+			if !item.Overridable.IsNull() {
+				itemBody, _ = sjson.Set(itemBody, "overridable", item.Overridable.ValueBool())
+			}
+			if !item.Ip.IsNull() {
+				itemBody, _ = sjson.Set(itemBody, "value", item.Ip.ValueString())
+			}
+			if !item.Type.IsNull() {
+				itemBody, _ = sjson.Set(itemBody, "type", item.Type.ValueString())
+			}
+			body, _ = sjson.SetRaw(body, "-1", itemBody)
+		}
+	}
+	return body
+}
+
+func (data *Hosts) fromBody(ctx context.Context, res gjson.Result) {
```

Comment:
Why is this section no longer generated/updated?

Reply:
I would like to get import working based on resources names, not their IDs. I think would make it easier to use. Hence condition was changed from:
```golang
if v.Get("id").String() == data.Id.ValueString() && data.Id.ValueString() != "" {
```
to
```golang
if v.Get("name").String() == k {
```

Reply:
This is template generated now